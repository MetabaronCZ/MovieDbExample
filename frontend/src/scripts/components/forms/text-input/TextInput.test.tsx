import { act } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';
import { fireEvent, getByRole } from '@testing-library/dom';

import { TextInput } from './TextInput';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/forms/TextInput', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <TextInput value="TEXT" onChange={() => null} />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const input = getByRole(container, 'textbox');
    expect(input).toBeInTheDocument();
    expect(input.tagName.toLowerCase()).toEqual('input');
    expect(input).toHaveAttribute('type', 'text');
    expect(input).toHaveAttribute('value', 'TEXT');
  });

  it('should render basic props', async () => {
    const { container } = await waitForComponent(
      <TextInput
        id="ID"
        value="TEXT"
        placeholder="PLACEHOLDER"
        disabled
        maxLength={120}
        onChange={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const input = getByRole(container, 'textbox');
    expect(input).toHaveAttribute('id', 'ID');
    expect(input).toHaveAttribute('value', 'TEXT');
    expect(input).toHaveAttribute('placeholder', 'PLACEHOLDER');
    expect(input).toHaveAttribute('disabled', '');
    expect(input).toHaveAttribute('maxlength', '120');
  });

  it('should be able to autofocus', async () => {
    const { container } = await waitForComponent(
      <TextInput value="TEXT" autoFocus onChange={() => null} />,
      { wrapper: TestComponentWrapper },
    );
    const input = getByRole(container, 'textbox');
    expect(input).toHaveFocus();
  });

  it('should be able to change value', async () => {
    const onChange = jest.fn();

    const { container } = await waitForComponent(
      <TextInput value="" onChange={onChange} />,
      { wrapper: TestComponentWrapper },
    );
    expect(onChange).not.toBeCalled();

    const input = getByRole(container, 'textbox');
    expect(input).toHaveAttribute('value', '');

    // type some text
    act(() => {
      fireEvent.change(input, { target: { value: 'TEXT' } });
    });

    expect(onChange).toBeCalledTimes(1);
    expect(onChange).toBeCalledWith('TEXT');
  });

  it('should be disabled', async () => {
    const onChange = jest.fn();

    const { container } = await waitForComponent(
      <TextInput value="TEXT" autoFocus disabled onChange={onChange} />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();
    expect(onChange).not.toBeCalled();

    const input = getByRole(container, 'textbox');
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('value', 'TEXT');
    expect(input).toHaveAttribute('disabled', '');
    expect(input).not.toHaveFocus(); // no autofocus when disabled

    // type some text
    act(() => {
      fireEvent.change(input, { target: { value: 'NEW_TEXT' } });
    });

    expect(onChange).not.toBeCalled();
  });
});
