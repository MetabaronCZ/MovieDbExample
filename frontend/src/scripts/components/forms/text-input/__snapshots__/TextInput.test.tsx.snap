// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/TextInput should be disabled 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: block;
  border: none;
  background: transparent;
  outline: none;
}

.c0::placeholder {
  font-style: italic;
  color: inherit;
  opacity: 0.5;
}

.c0:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c0:disabled {
  color: #98c3d9;
}

.c1 {
  width: 100%;
  padding: 0 12px;
  border: 2px solid #002d56;
  background: #fff;
  line-height: 1;
  height: 40px;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  border-color: #98c3d9;
}

<div>
  <input
    autocomplete="one-time-code"
    class="c0 c1"
    disabled=""
    type="text"
    value="TEXT"
  />
</div>
`;

exports[`components/forms/TextInput should render basic props 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: block;
  border: none;
  background: transparent;
  outline: none;
}

.c0::placeholder {
  font-style: italic;
  color: inherit;
  opacity: 0.5;
}

.c0:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c0:disabled {
  color: #98c3d9;
}

.c1 {
  width: 100%;
  padding: 0 12px;
  border: 2px solid #002d56;
  background: #fff;
  line-height: 1;
  height: 40px;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  border-color: #98c3d9;
}

<div>
  <input
    autocomplete="one-time-code"
    class="c0 c1"
    disabled=""
    id="ID"
    maxlength="120"
    placeholder="PLACEHOLDER"
    type="text"
    value="TEXT"
  />
</div>
`;

exports[`components/forms/TextInput should render properly 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: block;
  border: none;
  background: transparent;
  outline: none;
}

.c0::placeholder {
  font-style: italic;
  color: inherit;
  opacity: 0.5;
}

.c0:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c0:disabled {
  color: #98c3d9;
}

.c1 {
  width: 100%;
  padding: 0 12px;
  border: 2px solid #002d56;
  background: #fff;
  line-height: 1;
  height: 40px;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  border-color: #98c3d9;
}

<div>
  <input
    autocomplete="one-time-code"
    class="c0 c1"
    type="text"
    value="TEXT"
  />
</div>
`;
